### PR TITLE
tidy up shipped package

### PR DESCRIPTION
**Pull request by osher**
shipped tidy up:
```
{17:45}~/opensource/hyperid:master ✓ ➭ git checkout patch-2
Switched to branch 'patch-2'
Your branch is up to date with 'origin/patch-2'.
{17:45}~/opensource/hyperid:patch-2 ✓ ➭ npm pack
npm notice 
npm notice 📦  hyperid@2.0.3
npm notice === Tarball Contents === 
npm notice 1.1kB LICENSE     
npm notice 2.1kB hyperid.js  
npm notice 1.1kB package.json
npm notice 2.1kB README.md   
npm notice 456B  index.d.ts  
npm notice === Tarball Details === 
...
npm notice package size:  3.0 kB                                  
npm notice unpacked size: 6.8 kB   
```
vs, shipped unfiltered 
```
{17:42}~/opensource/hyperid:patch-2 ✓ ➭ git checkout master
Switched to branch 'master'
Your branch is up to date with 'origin/master'.
{17:43}~/opensource/hyperid:master ✓ ➭ npm pack
npm notice 
npm notice 📦  hyperid@2.0.3
npm notice === Tarball Contents === 
npm notice 1.1kB LICENSE           
npm notice 913B  benchmark.js      
npm notice 206B  example.js        
npm notice 2.1kB hyperid.js        
npm notice 1.9kB test/test.js      
npm notice 1.4kB test/uniqueness.js
npm notice 1.0kB package.json      
npm notice 154B  test/tsconfig.json
npm notice 2.1kB README.md         
npm notice 330B  test/example.ts   
npm notice 456B  index.d.ts        
npm notice 79B   .travis.yml       
npm notice === Tarball Details === 
...
npm notice package size:  4.2 kB                                  
npm notice unpacked size: 11.8 kB 
```

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -3,6 +3,10 @@
   "version": "2.0.3",
   "description": "Uber-fast unique id generation, for Node.js and the browser",
   "main": "hyperid",
+  "files": [
+    "hyperid.js",
+    "index.d.ts"
+  ],
```
Comment (mcollina) [inline package.json:9]: I’m not in favor of this. Tests are an integral part of this module.

**Comment by osher** [comment]
It does not make sense to me, please let me explain.

For me - part of how I evaluate a package is I check what's the size and if it ships files that do not belong to **production**.

Also - 

When a module is installed in `node_modules`, it is installed without it's dev-dependencies.
This means that you cannot _really_ run the tests, although you can read them.

If you'd expect for docs - what I'd do is provide links from within the `README.md` which by-definision, ships always.

In order to run the tests you have to `cd node_modules/hyperid && npm i` - which is something you're not encouraged to do. If you do it - you'll get `node_modules/hyperid/node_modules` which is different from what you get from `npm i` on your project root - in terms of duplicates and extranous dependencies. You're interfering with something that should have been encapsulated -  you basically should let npm manage it. - so It's an anti-pattern.

mm.
Can I please sincerely ask, do you really feel that people should run tests of a dependency from inside `node_modules`?


If you're not convinced - we can add `"tests",` to `"files": [`, but can we agree that all there rest do not belong in the package?
For sure - the `.travis-ci.yaml`...

File: package.json (hunk 1 above)
Comment (osher) [reply package.json:9]: It does not make sense to me, please let me explain.

For me - part of how I evaluate a package is I check what's the size and if it ships files that do not belong to production.

Also -

When a module is installed in node_modules, it is installed without it's dev-dependencies.
This means that you cannot really run the tests, although you can read them.

If you'd expect for docs - what I'd do is provide links from within the README.md which by-definision, ships always.

In order to run the tests you have to cd node_modules/hyperid && npm i - which is something you're not encouraged to do. If you do it - you'll get node_modules/hyperid/node_modules which is different from what you get from npm i on your project root - in terms of duplicates and extranous dependencies. You're interfering with something that should have been encapsulated - you basically should let npm manage it. - so It's an anti-pattern.

mm.
Can I please sincerely ask, do you really feel that people should run tests of a dependency from inside node_modules?

If you're not convinced - we can add "tests", to "files": [, but can we agree that all there rest do not belong in the package?
For sure - the .travis-ci.yaml...

Comment (mcollina) [reply package.json:9]: > Can I please sincerely ask, do you really feel that people should run tests of a dependency from inside node_modules?

Tests are a form of documentation. By looking at them, a developer can understand how a project is used. As a result, they are part of a release as much as docs is.

I'm pretty happy to have test in there and drop the rest.

Comment (osher) [reply package.json:9]: personally, I don't believe that docs belong in the package sent to production, but you da boss :grin:

**Comment by mcollina** [comment]
This adds a spurious tar.gz file.Can you remove that?

**Comment by osher** [comment]
OMG. shame on me.
well, this tar is the result of `npm pack` - which is how I check what will end up shipping.
In more complicated scenarios I even check how well it installs, and if it creates the `bin`s well, but now I just wanted to look at the shipped archive.

My bad - I missed it's being included in the `git add`. pitty...

**Pull request merged**